chapter02/worksheets/solution2.1.7.8.scala

```diff
@@ -0,0 +1,32 @@
+/** author: Andreas Röhler */
+
+/** 
+  Exercise 2.1.7.8
+
+  Write the solution of Exercise 2.1.7.7 as a function with type
+  parameters P and Q instead of the fixed types String and Int. Test
+  it with types P = Boolean and Q = Set[Int].
+ */
+
+/** 
+  Exercise 2.1.7.7
+
+  Given p:Seq[String] and q:Seq[Int] of equal length and assuming that
+  values in q do not repeat, compute a Map[Int, String] mapping numbers
+  from q to the corresponding strings from p.
+  */
+
+// def reorder[A](p: Seq[A], q: Seq[Int]): Seq[A] = ???
+def myMapping[P,Q](p: Seq[P], q: Seq[Q]): Map[Q, P] = {
+  val erg = q.zip(p).toMap
+  erg
+  }
+
+val result = myMapping(Seq("asdf", "bsdf", "csdf", "dsdf"), Seq(1, 2, 3, 4))
```

Comment:
Here you are testing with types `Int` and `String` because you are giving `Seq[Int]` and `Seq[String]` as arguments of `myMapping()`.

You can also test with `Seq[Set[Boolean]]` and `Seq[Double]` instead:

`myMapping( Seq( Set(true), Set(true, false), Set() ), Seq( 1.0, 2.0, 3.0))`